pkg/sql/plan2/function/builtin/unary/asin.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unary
+
+import (
+	"math"
+
+	"github.com/matrixorigin/matrixone/pkg/container/nulls"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/encoding"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+)
+
+type asinResult struct {
+	Result []float64
+	Nsp    *nulls.Nulls
+}
+
+func asin(inputValues []float64, rs []float64) asinResult {
+	result := asinResult{Result: rs, Nsp: new(nulls.Nulls)}
+	for i, n := range inputValues {
+		if n < -1 || n > 1 {
```

Comment:
As mentioned in [go asin implementation](https://go.dev/src/math/asin.go), math.asin handles the value greater than 1. Maybe, just call math.asin here and check the return value is nan or not. Same as acos.